### DISCUSSION TITLE
DTO Factory: tortoise implementation

### DISCUSSION THREAD

**Discussion by peterschutt** [litestar-org/litestar#1964] [General]
<a href="https://polar.sh/litestar-org/litestar/issues/1555">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/1555/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/litestar-org/litestar/issues/1555/pledge.svg">
</picture>
</a>

**Comment by Goldziher** [comment]
We removed the tortoise implementation of the plugin when we switched to DTO. The question now is whether we should re-implement this or not. I tend to think we should drop it - I am not aware of anyone who actually uses this, and tortoise does not seem to be very strongly maintained (unlike piccolo). 

Unless users chime in here and change our minds, we will simply keep this out of litestar.

**Reply by abhishek-compro** [reply]
I was thinking of using TortoiseORM but saw that the support was removed

**Reply by scottix** [reply]
It's still actively maintained despite the comment, actually it's not too hard to create a lifespan with it and use it in litestar. DTO support would be great to have as an additional support.

Honestly piccolo is very opinionated and found it quite hard to setup, maybe if I started with it from the beginning. Also found some features not supported yet, which made it a no go for me.

**Reply by enkelii6** [reply]
I wouldn't say that Tortoise isn't popular. Maybe not as popular as Alchemy, but I'm finding more and more enterprise code with Tortoise usage

Personally I prefer Tortoise over SQLAlchemy and it would be grate to see its support in litestar
